apps/console/src/features/identity-providers/components/settings/attribute-management/attribute-mapping-list-item.tsx

```diff
@@ -0,0 +1,301 @@
+import { FormApi } from "final-form";
+import React, { FunctionComponent, ReactElement, ReactNode, useEffect, useState } from "react";
+import { Field as FinalField, Form as FinalForm, FormRenderProps } from "react-final-form";
+import { Form as SemanticForm, Grid, Header, Input, Select } from "semantic-ui-react";
+import { IdentityProviderClaimInterface, IdentityProviderCommonClaimMappingInterface } from "../../../models";
+
+/**
+ * Props interface of {@link AttributeMappingListItem}
+ */
+export interface AttributeMappingListItemProps {
+    availableAttributeList: Array<IdentityProviderClaimInterface>;
+    editingMode?: boolean;
+    mapping?: IdentityProviderCommonClaimMappingInterface;
+    onSubmit: (mapping: IdentityProviderCommonClaimMappingInterface) => void;
+}
+
+/**
+ * This is a common interface that allows the user to map one attribute
+ * to another local attribute. The interface looks like this: -
+ *
+ * External Mapping:           Local Mappings List:
+ * +-----------------------+   +-----------------------+   +=====+
+ * | This is a text input  |   |  This is a dropdown   |   |  +  |
+ * +-----------------------+   +-----------------------+   +=====+
+ *                                       +=============+
+ *                                       | Add Mapping |
+ *                                       +=============+
+ *
+ * This component is a <Form> internally. And the fields value submissions
+ * are handled by onSubmit so, it is mandatory to have a submission
+ * button. If you pass editingMode={true} it will render plus icon button
+ * inline and hide button with text "Add Mapping Button" below it
+ * and input labels.
+ *
+ * @param props {AttributeMappingListIt emProps}
+ * @constructor
+ */
+export const AttributeMappingListItem: FunctionComponent<AttributeMappingListItemProps> = (
+    props: AttributeMappingListItemProps
+): ReactElement => {
+
+    const {
+        onSubmit,
+        availableAttributeList,
+        mapping,
+        editingMode
+    } = props;
+
+    const [ copyOfAttrs, setCopyOfAttrs ] = useState<Array<IdentityProviderClaimInterface>>([]);
+
+    useEffect(() => {
+        if (availableAttributeList) {
+            const copy = [ ...availableAttributeList ];
+            // When you enter into editing mode the available attribute list
+            // will not contain the mapping itself. We need to manually append
+            // it to the attrs to make it work.
+            if (editingMode && mapping && mapping.mappedValue && mapping.claim) {
+                copy.push(mapping.claim);
+            }
+            setCopyOfAttrs(copy);
+        }
+    }, [ availableAttributeList ]);
+
+    const _getListOfAvailableAttributes = () => {
```

Comment:
Using `_` as function name beginning is not a standard practice. Please fix.

Reply:
+1 Removed all the _ underscores from scoped functions. b49a285